### PR TITLE
Fix azure scaffolder git push auth

### PR DESCRIPTION
**Pull request by mattemoore**
## Hey, I just made a Pull Request!

     
Fixed issue https://github.com/backstage/backstage/issues/26368.
When Azure integration is configured to use App Registration or Managed Identity, the Azure:Publish action in Azure Devops Scaffolder module will be configured to authenticate to Git using basic authentication as opposed to Authorization header.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend-module-azure/src/actions/azure.ts
Comment (awanlin) [inline plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: Could we add a test for this in `azure.test.ts`, please?

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @mattemoore, would be great to add a test for this, especially seeing as there are some for this file already. 👍

File: plugins/scaffolder-backend-module-azure/src/actions/azure.ts
Comment (mattemoore) [reply plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: Sure. Give me a few hours.

Comment (awanlin) [reply plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: Navigate to the folder: `cd plugins/scaffolder-backend-module-azure/src/actions` and then run `yarn test azure.test.ts`

Comment (mattemoore) [reply plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: @awanlin from what I understand you'd like me to test that the proper format of auth is used when calling `initRepoAndPush`.  From what I can tell of the existing code to support the 2 configs (AppReg and ManagedIdentity) I would need to create multiple suites (describes).  Is that assumption correct and is that cool to have multiple suites in the same test file?

Comment (mattemoore) [reply plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: Sorry, i figured it out and deleted the comment.  For anyone else reading you can just do yarn test then pick the filter option.

Comment (mattemoore) [reply plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: Sorry  @awanlin  this may take me a bit longer.  It's been a minute since I've done mocking etc. in jest.  I'll need some time to relearn some things.

Comment (awanlin) [reply plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: That's alright, take the time you need. 👍

Comment (mattemoore) [reply plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: @awanlin i'm running into issues with the no-use-before-define ESLint rule.  My jest .spyOn()s are being hoisted above the import statements (as jest is want to do) but this results in the `no-use-before-define` rule being triggered.  Thoughts?

Comment (awanlin) [reply plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: I'm swamped and won't be able to help, testing is a bit of a weak spot for me so I would need to pull this down and play around with it.

Comment (awanlin) [reply plugins/scaffolder-backend-module-azure/src/actions/azure.ts:210]: OK, given @benjdlambert comment, let's leave it 👍

**Review by awanlin — APPROVED** [review]
Thanks again for running with this @mattemoore 🚀

**Comment by mattemoore** [comment]
Thanks.  I got my first if statement renewal contribution.  Anything else needed from my side?

**Comment by awanlin** [comment]
> Thanks. I got my first if statement renewal contribution. Anything else needed from my side?

Just need a review from the maintainers now, once they approve they will also merge. 👍

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**
